### PR TITLE
feat: Introduce non fiori3 page footer component

### PR DESCRIPTION
**Pull request by Lokanathan-k**
## Related Issue
Closes SAP/fundamental-styles#2232

## Description
introduce non fiori3 page footer component, as part of the platform ngx component development

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [NA] hover state of the element follow design spec
- [NA] focus state of the element follow design spec
- [NA] active state of the element follow design spec
- [NA] selected state of the element follow design spec
- [NA] selected hover state of the element follow design spec
- [NA] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [NA] Includes Compact/Cosy/Tablet design
- [X] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
- [x] last commit message should have `[ci visual]` so it can trigger chromatic visual regression (e.g. `test: run chromatic visual regression [ci visual]`)
4. Documentation
- [x] Storybook documentation has been created/updated
- [NA] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
1. Please provide link to the specs
2. Shouldn't there be some vertical spacing for the links?
Screen Shot 2021-04-07 at 3 09 23 PM

3. Is this the expected behaviour on smaller screens?

Screen Shot 2021-04-07 at 3 09 40 PM

5. Overlapping of text in smaller screens
Screen Shot 2021-04-07 at 3 08 59 PM

6. Shouldn't the image and the text of the link be aligned?
Screen Shot 2021-04-07 at 3 07 56 PM
Screen Shot 2021-04-07 at 3 07 47 PM

**Comment by InnaAtanasova** [comment]
Screen Shot 2021-04-07 at 3 13 20 PM

Added like this, the issue will not be closed automatically when the PR is merged.

File: src/page-footer.scss [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+  $fd-footer-padding-top: 0.75rem !default;
+  $fd-footer-margin-right: 1.5rem !default;
+  $fd-footer-link-padding-top: 0.25rem !default;
+
+  @mixin fd-truncate-text {
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+  }
```
Comment (InnaAtanasova) [inline src/page-footer.scss:24]: there's already a mixin for it. Please check the mixibs file

File: src/page-footer.scss [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    .#{$block}__text {
+      width: 90%;
+    }
+
+    .#{$block}__container {
+      display: inline-block;
+      flex-direction: row;
+    }
```
Comment (InnaAtanasova) [inline src/page-footer.scss:124]: all these rules are repeated in the other media query

File: stories/page-footer/page-footer.stories.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+			<span class="sap-icon--download sap-icon--s"></span>
+			 icon and link
+			</a>
+		</div>
+		<div class="fd-page-footer__row">
+			<a href="#" id="fd-link-14" class="fd-page-footer__row__item fd-page-footer__row__item--image fd-link" tabindex="0">
+			<img src="https://picsum.photos/id/1021/50/50" alt="page footer clickable image link"/>
+			image link
+			</a>
+			<a href="#" id="fd-link-15" class="fd-page-footer__row__item fd-page-footer__row__item--image fd-link" tabindex="0">
```
Comment (InnaAtanasova) [inline stories/page-footer/page-footer.stories.js:99]: `fd-page-footer__row__item` is not a BEM naming

File: src/page-footer.scss [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+  // sizes
+  @each $set-name, $size-set in $fd-page-footer-sizes {
+    &--#{$set-name} {
+      padding-right: map-get($size-set, 'padding');
+      padding-left: map-get($size-set, 'padding');
+      padding-top: 1.5rem;
+      padding-bottom: 2rem;
```
Comment (InnaAtanasova) [inline src/page-footer.scss:81]: these 2 should go on `fd-page-footer` level as they are the same for all cases, only the horizontal paddings are changing based on the modifier class.

File: src/page-footer.scss [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+      .#{$block}__container {
+        display: flex;
+
+        @include fd-flex(column);
```
Comment (InnaAtanasova) [inline src/page-footer.scss:85]: Have you checked what this mixin is doing? It brings `display: flex;` so you don't need code above this (line 83 where you have `display: flex;`

File: src/page-footer.scss [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+  "xl": ("padding": 3rem),
+  "lg": ("padding": 2rem),
+  "md": ("padding": 1rem),
+  "sm": ("padding": 0.5rem)
+);
+
+.#{$block} {
+  @include fd-reset();
+
+  $fd-footer-padding-top: 0.75rem !default;
```
Comment (InnaAtanasova) [inline src/page-footer.scss:16]: Used only once, no need to be a variable

File: src/page-footer.scss [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+  $fd-footer-margin-right: 1.5rem !default;
+  $fd-footer-link-padding-top: 0.25rem !default;
```
Comment (InnaAtanasova) [inline src/page-footer.scss:18]: Used only once, no need to be a variable

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

**Comment by InnaAtanasova** [comment]
@Lokanathan-k please don't resolve the comments, it's done by the reviewers. Please fix the PR template where you address the issue.

File: src/page-footer.scss (hunk 1 above)
Comment (Lokanathan-k) [reply src/page-footer.scss:24]: multiline-truncate() i see this mixin other than this there is no mixin available and this mixin add's unwanted margin on right. and there is no '...' at the end of the overflow text.
Screenshot 2021-04-14 at 11 47 41 AM

File: stories/page-footer/page-footer.stories.js [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+	</div>
+	<div class="fd-page-footer__container">
+		<div class="fd-page-footer__row">
+			<a href="#" id="fd-link-13"  class="fd-page-footer__row--item fd-link" tabindex="0">
+			<span class="sap-icon--download sap-icon--s"></span>
+			 icon and link
+			</a>
+		</div>
+		<div class="fd-page-footer__row">
+			<a href="#" id="fd-link-14" class="fd-page-footer__row--item fd-page-footer__row--item--image fd-link" tabindex="0">
```
Comment (InnaAtanasova) [inline stories/page-footer/page-footer.stories.js:95]: Please check the BEM naming!
`fd-page-footer__row--item--image` is not BEM. You can't have 2 modifiers.
`fd-page-footer` is your BLOCK
`fd-page-footer__row` is your ELEMENT, that's why it has `__`
`fd-page-footer__row--item` is your modifier, that's why you have `--`

But you can't add a second modifier or a second element. 
You need to change `fd-page-footer__row--item` to `fd-page-footer__row-item`, in this case `__row-item` is your element and then have `fd-page-footer__row-item--image` as a modifier.

File: src/page-footer.scss [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+    &--item {
+      margin-right: $fd-footer-margin-right;
+
+      @include fd-rtl() {
+        margin-right: 0;
+        margin-left: $fd-footer-margin-right;
+      }
+
+      &--image {
+        display: inline-grid;
```
Comment (InnaAtanasova) [inline src/page-footer.scss:47]: Please reconsider this. Grid is not working in IE11 and we need to support it.

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

File: stories/page-footer/page-footer.stories.js [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+	<div class="fd-page-footer__container">
+		<div class="fd-page-footer__row">
+			<a href="#" id="fd-link-13"  class="fd-page-footer__row-item fd-link" tabindex="0">
+			<span class="sap-icon--download sap-icon--s"></span>
+			 icon and link
+			</a>
+		</div>
+		<div class="fd-page-footer__row">
+			<a href="#" id="fd-link-14" class="fd-page-footer__row-item fd-link" tabindex="0">
+			<img class="fd-page-footer__row-item--image" src="https://picsum.photos/id/1021/50/50" alt="page footer clickable image link"/>
```
Comment (InnaAtanasova) [inline stories/page-footer/page-footer.stories.js:96]: What are you modifying with a modifier class if you don't have a base class?

You need the following format:
```
class="fd-block__element fd-block__element--modifier"
```

You can't have just:
```
class="fd-block__element--modifier"
```

A modifier class is used to change a property/properties of a base class

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Review by sKudum — APPROVED** [review]

**Pull request merged**